controllers/backup_controller.go

```diff
@@ -0,0 +1,114 @@
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package controllers
+
+import (
+	"context"
+	"fmt"
+	"reflect"
+
+	"github.com/presslabs/controller-util/syncer"
+	"k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/client-go/tools/record"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/log"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+
+	apiv1alpha1 "github.com/radondb/radondb-mysql-kubernetes/api/v1alpha1"
+	"github.com/radondb/radondb-mysql-kubernetes/backup"
+	backupSyncer "github.com/radondb/radondb-mysql-kubernetes/backup/syncer"
+)
+
+// BackupReconciler reconciles a Backup object
+type BackupReconciler struct {
+	client.Client
+	Scheme   *runtime.Scheme
+	Recorder record.EventRecorder
+}
+
+//+kubebuilder:rbac:groups=mysql.radondb.com,resources=backups,verbs=get;list;watch;create;update;patch;delete
+//+kubebuilder:rbac:groups=batch,resources=jobs,verbs=get;list;watch;create;update;patch;delete
+//+kubebuilder:rbac:groups=mysql.radondb.com,resources=backups/status,verbs=get;update;patch
+
+// Reconcile is part of the main kubernetes reconciliation loop which aims to
+// move the current state of the cluster closer to the desired state.
+// TODO(user): Modify the Reconcile function to compare the state specified by
+// the Backup object against the actual cluster state, and then
+// perform operations to make the cluster state reflect the state specified by
+// the user.
+//
+// For more details, check Reconcile and its Result here:
+// - https://pkg.go.dev/sigs.k8s.io/controller-runtime@v0.7.2/pkg/reconcile
+func (r *BackupReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Result, error) {
+	// your logic here
+	// Fetch the Backup instance
+	backup := backup.New(&apiv1alpha1.Backup{})
+	err := r.Get(context.TODO(), req.NamespacedName, backup.Unwrap())
+	if err != nil {
+		if errors.IsNotFound(err) {
+			// Object not found, return.  Created objects are automatically garbage collected.
+			// For additional cleanup logic use finalizers.
+			return reconcile.Result{}, nil
+		}
+		// Error reading the object - requeue the request.
+		return reconcile.Result{}, err
+	}
+	// Set defaults on backup
+	r.Scheme.Default(backup.Unwrap())
+
+	// save the backup for later check for diff
+	savedBackup := backup.Unwrap().DeepCopy()
+
+	jobSyncer := backupSyncer.NewJobSyncer(r.Client, r.Scheme, backup)
+	if err := syncer.Sync(ctx, jobSyncer, r.Recorder); err != nil {
+		return reconcile.Result{}, err
+	}
+
+	if err = r.updateBackup(savedBackup, backup); err != nil {
+		return reconcile.Result{}, err
+	}
+
+	return ctrl.Result{}, nil
+}
+
+// SetupWithManager sets up the controller with the Manager.
+func (r *BackupReconciler) SetupWithManager(mgr ctrl.Manager) error {
+	return ctrl.NewControllerManagedBy(mgr).
```

Comment:
return ctrl.NewControllerManagedBy(mgr).
		For(&apiv1alpha1.Backup{}).
                Owns(&batchv1.Job{}).
		Complete(r)